aten/src/ATen/native/cuda/Blas.cu

```diff
@@ -0,0 +1,53 @@
+#include <ATen/cuda/CUDABlas.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <c10/macros/Macros.h>
+
+namespace at {
+namespace cuda {
+namespace blas {
+
+  template <typename scalar_t>
+  __global__ void gemm_kernel(int64_t M, int64_t N, int64_t K, scalar_t alpha, const scalar_t* a, int64_t lda,
```

Comment:
Since this is a new file you have an opportunity to use a more descriptive name, something like "IntegerMatmul.cu" instead of vague "Blas". Maybe "IntegerBlas" if the plan is to implement integer support also for other functions from BLAS API.

Reply:
Since there's no plan for other integer BLAS operations I'll confine it to IntegerMatmul.cu for now.